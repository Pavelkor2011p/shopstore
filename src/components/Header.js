import React from 'react'

export default function Header() {
    return (
    <header className="header">
    <div className="header__firstline  conteiner">
        <div className="header__firstline-left">
            <div className='header__firstline-left_logo'></div>
            <p className="header__firstline-left_adress"><strong>г. Кокчетав, ул. Ж. Ташенова 129Б</strong><br /><span className="header__firstline-left_west">(Рынок Восточный)</span></p>
            <div className="header__firstline-left">
                <div className='header__firstline-left_email'></div>
                <p className="header__firstline-left_adress"><strong>opt.sultan@mail.ru</strong><br />
                <span className="header__firstline-left_west">На связи в любое время</span></p>
            </div>
        </div>
    <div className="header__firstline-right">
        <nav >
        <ul className="header__firstline-nav">
            <li><a href="/">О компании</a></li>
            <li><a href="/">Доставка и оплата</a></li>
            <li><a href="/">Возврат</a></li>
            <li><a href="/">Контакты</a></li>
            </ul>
        </nav>
    </div>
    </div>
    <div className="header__wrapp">
    <div className="conteiner header__secondline">
        <div className="header__secondline-logo"></div>
        <button className="header__secondline-button button">Каталог  <div className='catalog'></div></button>
            <div className="header__secondline-input">
                <input type="search" placeholder="Поиск..." className="header__secondline-input_search" />
                <input type="button" className="header__secondline-input_button" />
            </div>
        <div className="header__secondline-fone">
            <p><strong>+7 (777) 490-00-91</strong></p>
            <p><span className="header__secondline-fone_west">время работы: 9:00-20:00</span></p><br />
            <a href="tel:+7 (777) 490-00-91" className="header__secondline_tel">Заказать звонок</a>
        </div>
        <div className="header__secondline-img"></div>
        <button className="header__secondline-button button">Прайс-лист <div className='download'></div></button>
        <div className="header__secondline-basket">
            <button className="header__secondline-basket_button"></button>
            <div>
                <p>Корзина</p>
            </div>
        </div>
    </div>
    </div>
</header>
    )
}
