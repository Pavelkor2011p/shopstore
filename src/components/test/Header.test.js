
import{render, screen} from '@testing-library/react';
import Header from "../Header";

describe('Header', () => {
    it('should create Header with empty header', () => {
        const utils = render(<Header />);

        expect(utils).toMatchSnapshot();
    });

    it('list renders', ()=> {
        render(<Header />);
        const Element = screen.getAllByText(/оплата/i)

        expect(Element).toMatchSnapshot();
    })

    it('list render', ()=> {
        render(<Header />);
        const Elements = screen.getAllByText(/компании/i)
        expect(Elements).toMatchSnapshot();
    })

    it('title render', ()=> {
        render(<Header />);
        const Elements = screen.getAllByText(/Кокчетав/i)
        expect(Elements).toMatchSnapshot();
    })
})