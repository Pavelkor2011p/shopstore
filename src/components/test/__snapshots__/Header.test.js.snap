// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header list render 1`] = `
Array [
  <a
    href="/"
  >
    О компании
  </a>,
]
`;

exports[`Header list renders 1`] = `
Array [
  <a
    href="/"
  >
    Доставка и оплата
  </a>,
]
`;

exports[`Header should create Header with empty header 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <div
          class="header__firstline  conteiner"
        >
          <div
            class="header__firstline-left"
          >
            <div
              class="header__firstline-left_logo"
            />
            <p
              class="header__firstline-left_adress"
            >
              <strong>
                г. Кокчетав, ул. Ж. Ташенова 129Б
              </strong>
              <br />
              <span
                class="header__firstline-left_west"
              >
                (Рынок Восточный)
              </span>
            </p>
            <div
              class="header__firstline-left"
            >
              <div
                class="header__firstline-left_email"
              />
              <p
                class="header__firstline-left_adress"
              >
                <strong>
                  opt.sultan@mail.ru
                </strong>
                <br />
                <span
                  class="header__firstline-left_west"
                >
                  На связи в любое время
                </span>
              </p>
            </div>
          </div>
          <div
            class="header__firstline-right"
          >
            <nav>
              <ul
                class="header__firstline-nav"
              >
                <li>
                  <a
                    href="/"
                  >
                    О компании
                  </a>
                </li>
                <li>
                  <a
                    href="/"
                  >
                    Доставка и оплата
                  </a>
                </li>
                <li>
                  <a
                    href="/"
                  >
                    Возврат
                  </a>
                </li>
                <li>
                  <a
                    href="/"
                  >
                    Контакты
                  </a>
                </li>
              </ul>
            </nav>
          </div>
        </div>
        <div
          class="header__wrapp"
        >
          <div
            class="conteiner header__secondline"
          >
            <div
              class="header__secondline-logo"
            />
            <button
              class="header__secondline-button button"
            >
              Каталог  
              <div
                class="catalog"
              />
            </button>
            <div
              class="header__secondline-input"
            >
              <input
                class="header__secondline-input_search"
                placeholder="Поиск..."
                type="search"
              />
              <input
                class="header__secondline-input_button"
                type="button"
              />
            </div>
            <div
              class="header__secondline-fone"
            >
              <p>
                <strong>
                  +7 (777) 490-00-91
                </strong>
              </p>
              <p>
                <span
                  class="header__secondline-fone_west"
                >
                  время работы: 9:00-20:00
                </span>
              </p>
              <br />
              <a
                class="header__secondline_tel"
                href="tel:+7 (777) 490-00-91"
              >
                Заказать звонок
              </a>
            </div>
            <div
              class="header__secondline-img"
            />
            <button
              class="header__secondline-button button"
            >
              Прайс-лист 
              <div
                class="download"
              />
            </button>
            <div
              class="header__secondline-basket"
            >
              <button
                class="header__secondline-basket_button"
              />
              <div>
                <p>
                  Корзина
                </p>
              </div>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <div
        class="header__firstline  conteiner"
      >
        <div
          class="header__firstline-left"
        >
          <div
            class="header__firstline-left_logo"
          />
          <p
            class="header__firstline-left_adress"
          >
            <strong>
              г. Кокчетав, ул. Ж. Ташенова 129Б
            </strong>
            <br />
            <span
              class="header__firstline-left_west"
            >
              (Рынок Восточный)
            </span>
          </p>
          <div
            class="header__firstline-left"
          >
            <div
              class="header__firstline-left_email"
            />
            <p
              class="header__firstline-left_adress"
            >
              <strong>
                opt.sultan@mail.ru
              </strong>
              <br />
              <span
                class="header__firstline-left_west"
              >
                На связи в любое время
              </span>
            </p>
          </div>
        </div>
        <div
          class="header__firstline-right"
        >
          <nav>
            <ul
              class="header__firstline-nav"
            >
              <li>
                <a
                  href="/"
                >
                  О компании
                </a>
              </li>
              <li>
                <a
                  href="/"
                >
                  Доставка и оплата
                </a>
              </li>
              <li>
                <a
                  href="/"
                >
                  Возврат
                </a>
              </li>
              <li>
                <a
                  href="/"
                >
                  Контакты
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
      <div
        class="header__wrapp"
      >
        <div
          class="conteiner header__secondline"
        >
          <div
            class="header__secondline-logo"
          />
          <button
            class="header__secondline-button button"
          >
            Каталог  
            <div
              class="catalog"
            />
          </button>
          <div
            class="header__secondline-input"
          >
            <input
              class="header__secondline-input_search"
              placeholder="Поиск..."
              type="search"
            />
            <input
              class="header__secondline-input_button"
              type="button"
            />
          </div>
          <div
            class="header__secondline-fone"
          >
            <p>
              <strong>
                +7 (777) 490-00-91
              </strong>
            </p>
            <p>
              <span
                class="header__secondline-fone_west"
              >
                время работы: 9:00-20:00
              </span>
            </p>
            <br />
            <a
              class="header__secondline_tel"
              href="tel:+7 (777) 490-00-91"
            >
              Заказать звонок
            </a>
          </div>
          <div
            class="header__secondline-img"
          />
          <button
            class="header__secondline-button button"
          >
            Прайс-лист 
            <div
              class="download"
            />
          </button>
          <div
            class="header__secondline-basket"
          >
            <button
              class="header__secondline-basket_button"
            />
            <div>
              <p>
                Корзина
              </p>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header title render 1`] = `
Array [
  <strong>
    г. Кокчетав, ул. Ж. Ташенова 129Б
  </strong>,
]
`;
